### ISSUE TITLE
[false-positive] disconected foreach statement with self-assignment

### ISSUE THREAD

**Issue by kalessil** [kalessil/phpinspectionsea#22]
Original link: https://bitbucket.org/kalessil/phpinspectionsea/issues/237/disconnectedforeachinstructioninspection
Author @nazar-pc

```php
$files = [];
$hash  = 'x';
foreach ($files as &$file) {
	$file .= "?$hash";
}
```

Variations like `$file .= "?x"` and `$file = "$file?$hash"` are handled correctly
